File: code/python/wget.1.py
#!/usr/bin/python
# writen by andrewt@cse.unsw.edu.au as a COMP2041 example
# fetch a web page remove HTML tags and constants
# and print non-empty lines
#
# The regex code below doesn't handle a number of cases.  It is often
# better to use a library to properly parse HTML before processing it.
# But beware illegal HTML is common & often causes problems for parsers.

import re, sys

# urllib package names changed in Python 3
try:
    from urllib import urlopen # Python 2
except ImportError:
    from urllib.request import urlopen # Python 3

for url in sys.argv[1:]:
    response = urlopen(url)
    webpage = response.read().decode()
    webpage = re.sub(r'(?i)<script>.*?</script>', '', webpage)
    webpage = re.sub(r'(?i)<style>.*?</style>', '', webpage)
    webpage = re.sub(r'&\w+;', ' ', webpage)
    webpage = re.sub(r'<[^>]*>', '', webpage)
    webpage = re.sub(r'\n\s*\n', '\n', webpage)
    sys.stdout.write(webpage)

File: code/perl_regex/wget.1.pl
#!/usr/bin/perl -w
# written by andrewt@cse.unsw.edu.au as a COMP2041 lecture example
# Fetch a web page removing HTML tags and constants
# The contents of script or style tags are removed..
# Non-blank lines are printed
#
# The regex code below doesn't handle a number of cases.  It is often
# better to use a library to properly parse HTML before processing it.
# But beware illegal HTML is common & often causes problems for parsers.
#
# note the use of the s modifier to allow . to match a newline
#


use LWP::Simple;
foreach $url (@ARGV) {
	$html = get $url;
	$html =~ s/<script.*?<\/script>//isg;  # remove script tags including contents
	$html =~ s/<style.*?<\/style>//isg;    # remove style tags including contents
    $html =~ s/<.*?>//isg; # remove tags
    $html =~ s/\n\s*\n/\n/ig;  # blank lines
    print $html;
}
